### PR TITLE
[w5.7][T12-1]Winson Heng Yong Seng

### PR DESCRIPTION
**Pull request by winsonhys**
Added sort command and updated user doc

### CODE REVIEW

**Comment by JasonChong96** [comment]
Nice command

**Comment by JasonChong96** [comment]
Very useful

File: .classpath [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (parvathy-sudhir-pillai) [inline .classpath:1]: This change does not seem to be related to the enhancement proposed by this PR 🤔

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -67,6 +67,20 @@ Examples:
 * `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
 * `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
 
+== Sorting the list: `sort`
+
+Sorts the list based on a field +
+Format: `add FIELD`
```
Comment (parvathy-sudhir-pillai) [inline docs/UserGuide.adoc:73]: Wrong command?

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.Person;
+
+import java.util.Comparator;
+
+public class SortCommand extends Command {
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:7]: Coding standard requires to include descriptive header comments for public classes

File: src/seedu/addressbook/commands/SortCommand.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    private String args;
+
+    public static final String COMMAND_WORD = "sort";
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:11]: Try to avoid public class member variables

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/data/person/UniquePersonList.java:3]: Coding standard does not permit wildcard imports

File: src/seedu/addressbook/parser/Parser.java [hunk 6]
```diff
@@ -73,33 +74,37 @@ public Command parseCommand(String userInput) {
 
         switch (commandWord) {
 
-        case AddCommand.COMMAND_WORD:
-            return prepareAdd(arguments);
+            case SortCommand.COMMAND_WORD:
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/parser/Parser.java:77]: Wrong indentation? Our coding standard requires 4 spaces (instead of tabs) for normal indentation and an indentation of 8 spaces for line wrapping (i.e. when breaking a long line into multiple lines).

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.commands;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.*;
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:7]: Wildcard import not permitted

File: test/input.txt [hunk 8]
```diff
@@ -140,6 +140,14 @@
   delete 1
   list
 
+##########################################################
+# test exit command
+##########################################################
+
+  # sorts properly
+  sort name
```
Comment (parvathy-sudhir-pillai) [inline test/input.txt:148]: You may want to change the expected.txt to reflect the results of this command

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.util.TestUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+public class SortCommandTest {
+
+
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:15]: Stray line break? Try to avoid these spaces by checking diff while submitting a PR.

File: test/java/seedu/addressbook/commands/SortCommandTest.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        Person davidGrant = new Person(new Name("David Grant"), new Phone("61121122", false),
+                new Email("david@grant.com", false), new Address("44H Define Road", false),
+                Collections.emptySet());
+
+        emptyAddressBook = TestUtil.createAddressBook();
+        addressBook = TestUtil.createAddressBook(johnDoe, janeDoe, davidGrant, samDoe);
+    }
+
+    @Test
+    public void correctlySort() {
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/commands/SortCommandTest.java:36]: Coding standard requires to include descriptive header comments for public methods

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 11]
```diff
@@ -43,4 +43,13 @@ private void assertAreUnique(Object... objects) {
     private void assertNotUnique(Object... objects) {
         assertFalse(Utils.elementsAreUnique(Arrays.asList(objects)));
     }
+
+    @Test
+    public void isAnyNull() throws Exception {
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:48]: This commit shouldn't be part of this PR. In general, a PR should not contain unmerged code from other PRs. Try to limit each PR to one enhancement, which means the PR should be based on a branch that starts off from the master branch that is in sync with the upstream master branch

**Review by parvathy-sudhir-pillai — APPROVED** [review]
Good effort on this PR @winsonhys. 

Suggestions to improve further:
* update user and dev docs
* update jUnit and I/O tests to test the new functionality
* limit the PR to one enhancement and avoid including changes not related to that enhancement or unmerged code from other PRs
* pay closer attention to the coding standard
* ensure codebase builds fully on IntelliJ and I/O tests pass 

Please close the PR after viewing comments.
